plugins/module_utils/common.py

```diff
@@ -99,201 +100,107 @@
     K8S_IMP_ERR = traceback.format_exc()
 
 
-def list_dict_str(value):
-    if isinstance(value, (list, dict, string_types)):
-        return value
-    raise TypeError
-
-
-ARG_ATTRIBUTES_BLACKLIST = ('property_path',)
-
-COMMON_ARG_SPEC = {
-    'state': {
-        'default': 'present',
-        'choices': ['present', 'absent'],
-    },
-    'force': {
-        'type': 'bool',
-        'default': False,
-    },
-}
-
-RESOURCE_ARG_SPEC = {
-    'resource_definition': {
-        'type': list_dict_str,
-        'aliases': ['definition', 'inline']
-    },
-    'src': {
-        'type': 'path',
-    },
-}
-
-NAME_ARG_SPEC = {
-    'kind': {},
-    'name': {},
-    'namespace': {},
-    'api_version': {
-        'default': 'v1',
-        'aliases': ['api', 'version'],
-    },
-}
-
-AUTH_ARG_SPEC = {
-    'kubeconfig': {
-        'type': 'path',
-    },
-    'context': {},
-    'host': {},
-    'api_key': {
-        'no_log': True,
-    },
-    'username': {},
-    'password': {
-        'no_log': True,
-    },
-    'validate_certs': {
-        'type': 'bool',
-        'aliases': ['verify_ssl'],
-    },
-    'ca_cert': {
-        'type': 'path',
-        'aliases': ['ssl_ca_cert'],
-    },
-    'client_cert': {
-        'type': 'path',
-        'aliases': ['cert_file'],
-    },
-    'client_key': {
-        'type': 'path',
-        'aliases': ['key_file'],
-    },
-    'proxy': {
-        'type': 'str',
-    },
-    'persist_config': {
-        'type': 'bool',
-    },
-}
-
-WAIT_ARG_SPEC = dict(
-    wait=dict(type='bool', default=False),
-    wait_sleep=dict(type='int', default=5),
-    wait_timeout=dict(type='int', default=120),
-    wait_condition=dict(
-        type='dict',
-        default=None,
-        options=dict(
-            type=dict(),
-            status=dict(type='str', default="True", choices=["True", "False", "Unknown"]),
-            reason=dict()
-        )
-    )
-)
-
-DELETE_OPTS_ARG_SPEC = {
-    'propagationPolicy': {
-        'choices': ['Foreground', 'Background', 'Orphan'],
-    },
-    'gracePeriodSeconds': {
-        'type': 'int',
-    },
-    'preconditions': {
-        'type': 'dict',
-        'options': {
-            'resourceVersion': {
-                'type': 'str',
-            },
-            'uid': {
-                'type': 'str',
-            }
-        }
-    }
-}
-
-
-# Map kubernetes-client parameters to ansible parameters
-AUTH_ARG_MAP = {
-    'kubeconfig': 'kubeconfig',
-    'context': 'context',
-    'host': 'host',
-    'api_key': 'api_key',
-    'username': 'username',
-    'password': 'password',
-    'verify_ssl': 'validate_certs',
-    'ssl_ca_cert': 'ca_cert',
-    'cert_file': 'client_cert',
-    'key_file': 'client_key',
-    'proxy': 'proxy',
-    'persist_config': 'persist_config',
-}
-
+def configuration_digest(configuration):
+    import hashlib
+    m = hashlib.sha256()
+    for k in AUTH_ARG_MAP:
```

Comment:
I'm still unclear on which python versions we are supporting. If the intent is for this to work on 2.7 then I think we should use an OrderedDict to ensure the caching works. If not, feel free to ignore.

Reply:
Since we will only cache the session on py3.6+, so we can keep the code as it is today.